ompi/mca/pml/cm/pml_cm_recvreq.h

```diff
@@ -134,6 +137,7 @@ do {                                                                    \
     OBJ_RETAIN(comm);                                                   \
     OMPI_DATATYPE_RETAIN(datatype);                                     \
                                                                         \
+    MCA_PML_CM_SWITCH_CUDA_CONVERTOR_OFF(ompi_mpi_local_convertor);     \
```

Comment:
I don't think you should meddle with the global convertor, nor in fact with any convertor that does not belong to the MTL requests. I noticed that you also alter the ```ompi_proc->super.proc_convertor```, which is used as a master copy for the convertor and will be copied by all convertors for the peer.

Thus this patch might work for simple blocking applications, but it will not work for any multi-threaded scenarios (as you alter a convertor that will be used by multiple requests in same time instead of altering the convertor that belongs to the request).

Reply:
Hi @bosilca,  just be on the same page, you mean the flags should be set on ```&(request)->req_base.req_convertor``` , right?

Reply:
Yes, ```(request)->req_base.req_convertor``` is the convertor associate with each request.

Reply:
ok, will work with @aravindksg to update the patch. thanks!